src/main/java/step1/Calculator.java

```diff
@@ -0,0 +1,25 @@
+package step1;
+
+import java.util.List;
+
+public class Calculator {
+
+    private final List<Integer> numbers;
+    private final List<Operator> operators;
+
+    public Calculator(List<Integer> numbers, List<Operator> operators) {
+        this.numbers = numbers;
+        this.operators = operators;
+    }
+
+    public int calculate() {
+        int result = numbers.get(0);
+        for (int i = 0; i < operators.size(); i++) {
+            Operator operator = operators.get(i);
+            Integer rightNumber = numbers.get(i + 1);
+
+            result = operator.apply(result, rightNumber);
+        }
+        return result;
+    }
```

Comment:
> Calculator.calculate() 부분 : 위와 비슷한 질문인데, 반복문 안의 코드가 다른 사람이 보기에 명시적이지 않은 것 같아 변화를 어떻게 줄 지, 줘야할지 고민이 있었습니다..

이 부분에 대해 저도 고민을 해보았는데요..!
`get(i)` 으로 index에 해당하는 값을 찾아오는것도 좋지만, 큐를 사용하면 index 없이 구현할 수 있을것 같아요!

```suggestion
    public int calculate() {
        int result = numbers.pop();

        while (!numbers.isEmpty()) {
            Integer n = numbers.pop();
            Operator operator = operators.pop();
            result = operator.apply(result, n);
        }
        
        return result;
    }
```

Reply:
Queue를 쓸 생각을 아예 못했네요.. 엄청 코드가 간결해지고 보기 좋은 것 같아요.
알고리즘에만 적용하다가 이렇게 적용할 생각은 못했네요ㅜㅜ 제안 감사드립니다😃